src/kaltura-player.ts

```diff
@@ -170,12 +170,34 @@ export class KalturaPlayer extends FakeEventTarget {
       this.setMedia(mediaConfig);
       return mediaConfig;
     } catch (e) {
-      const error = new Error(Error.Severity.CRITICAL, Error.Category.PLAYER, Error.Code.LOAD_FAILED, e);
+      const category = this._getErrorCategory(e);
+      const error = new Error(Error.Severity.CRITICAL, category, Error.Code.LOAD_FAILED, e);
       this._localPlayer.dispatchEvent(new FakeEvent(CoreEventType.ERROR, error));
       throw e;
     }
   }
 
+  private _getErrorCategory(error: Error): number {
```

Comment:
I think its hard to make a dictionary out of it since its has multiple layer of decisions, but we sure need to move the code of this function to another file